src/commonMain/kotlin/org/mongodb/kbson/ExperimentalKSerializerApi.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mongodb.kbson
+
+@MustBeDocumented
+@Target(
+    AnnotationTarget.CLASS,
+    AnnotationTarget.PROPERTY,
+    AnnotationTarget.FUNCTION,
+    AnnotationTarget.TYPEALIAS
+)
+@RequiresOptIn(level = RequiresOptIn.Level.ERROR)
+/**
+ * Marks declarations that use **experimental** APIs from `kotlinx.serialization` in `org.mongodb.kbson`.
+ * It means that the corresponding declarations might have issues due to incompatible versions when
+ * using a different one than the one used by KBson, or open issues in `kserializer`.
+ */
+public annotation class ExperimentalKSerializerApi
```

Comment:
I added it because the Kbson is already in the annotations package name, but you are right that it is more explicit to have it in the annotation itself, it helps to identify what makes the function to be experimental.